tests/unit/s2n_client_key_share_extension_test.c

```diff
@@ -165,11 +164,40 @@ int main(int argc, char **argv)
     /* Test s2n_client_key_share_extension.recv */
     {
         /* Test that s2n_client_key_share_extension.recv is a no-op
-         * if tls1.3 not enabled */
+         * if tls1.3 not enabled AND in use  */
         {
+            struct s2n_connection *client_conn, *server_conn;
+            EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+            EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+
+            DEFER_CLEANUP(struct s2n_stuffer key_share_extension, s2n_stuffer_free);
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&key_share_extension, 0));
+
+            EXPECT_SUCCESS(s2n_client_key_share_extension.send(client_conn, &key_share_extension));
+            uint16_t key_share_extension_size = s2n_stuffer_data_available(&key_share_extension);
+
+            EXPECT_SUCCESS(s2n_disable_tls13());
+            server_conn->actual_protocol_version = S2N_TLS12;
+            EXPECT_SUCCESS(s2n_client_key_share_extension.recv(server_conn, &key_share_extension));
+            EXPECT_EQUAL(s2n_stuffer_data_available(&key_share_extension), key_share_extension_size);
```

Comment:
If key_share extensions are present only for TLS1.3, why is s2n_client_key_share_extension.recv passing for TLS1.2 here?

Reply:
It's a no-op in TLS1.2. We don't process the extension. The check is that we don't actually read any of the extension data from the stuffer.